apps/extension/src/routes/page/onboarding/set-rpc-endpoint.tsx

```diff
@@ -0,0 +1,94 @@
+import { Card, CardDescription, CardHeader, CardTitle } from '@penumbra-zone/ui/components/ui/card';
+import { FadeTransition } from '@penumbra-zone/ui/components/ui/fade-transition';
+import { RPC_ENDPOINTS } from '../../../shared/rpc-endpoints';
+import { FormEvent, useMemo, useRef, useState } from 'react';
+import { SelectList } from '@penumbra-zone/ui/components/ui/select-list';
+import { Button } from '@penumbra-zone/ui/components/ui/button';
+import { useStore } from '../../../state';
+import { usePageNav } from '../../../utils/navigate';
+import { PagePath } from '../paths';
+import { ServicesMessage } from '@penumbra-zone/types/src/services';
+import { Network } from 'lucide-react';
+
+const randomSort = () => (Math.random() >= 0.5 ? 1 : -1);
+
+export const SetRpcEndpoint = () => {
+  const navigate = usePageNav();
+  const randomlySortedEndpoints = useMemo(() => [...RPC_ENDPOINTS].sort(randomSort), []);
+  const [grpcEndpoint, setGrpcEndpoint] = useState(randomlySortedEndpoints[0]!.url);
+  const setGrpcEndpointInZustand = useStore(state => state.network.setGRPCEndpoint);
+  const customRpcEndpointInput = useRef<HTMLInputElement | null>(null);
+  const isCustomRpcEndpoint = !RPC_ENDPOINTS.some(({ url }) => url === grpcEndpoint);
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    await setGrpcEndpointInZustand(grpcEndpoint);
+    void chrome.runtime.sendMessage(ServicesMessage.OnboardComplete);
+    navigate(PagePath.ONBOARDING_SUCCESS);
+  };
+
+  return (
+    <FadeTransition>
+      <Card className='w-[400px]' gradient>
+        <CardHeader>
+          <CardTitle>Select your preferred RPC endpoint</CardTitle>
+          <CardDescription>
+            If you&apos;re unsure which one to choose, leave this option set to the default.
+          </CardDescription>
```

Comment:
Can we add some text about the importance of trusting this RPC as it reveals intentions by what you request from it?

Reply:
done